nanofi/include/core/cstructs.h

```diff
@@ -155,9 +155,33 @@ typedef struct cstream {
  * ##################################################################
  */
 
-typedef struct tokens {
-    char ** str_list;
-    uint64_t num_strings;
+typedef struct token_node {
+    char * data;
+    struct token_node * next;
+} token_node;
+
+typedef struct token_list {
```

Comment:
yes, I remember that and I mentioned in that PR that this token_list has some additional members that was important for the log_aggregator ecu. If I use utlist I will have to have some other structure outside of the list, to keep track of the offset in the file. Currently the token_list encapsulates that so that token nodes and the offset stay together and it is better to keep it that way.

Reply:
Yes, I remember that and I mentioned in that PR that if I use utlist, I will have to use an external structure to keep track of extra variables. Currently it is encapsulated in token_list staying together with the tokens. Using utlist would make these variables to be unnecessarily copied into each linked list node.